server/sonar-server/src/test/java/org/sonar/server/favorite/ws/AddActionTest.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.favorite.ws;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.web.UserRole;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.DbTester;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.property.PropertyDto;
+import org.sonar.db.property.PropertyQuery;
+import org.sonar.server.component.ComponentFinder;
+import org.sonar.server.component.FavoriteService;
+import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.server.exceptions.UnauthorizedException;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.TestRequest;
+import org.sonar.server.ws.TestResponse;
+import org.sonar.server.ws.WsActionTester;
+
+import static java.net.HttpURLConnection.HTTP_NO_CONTENT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.core.util.Protobuf.setNullable;
+import static org.sonar.db.component.ComponentTesting.newFileDto;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+import static org.sonarqube.ws.client.favorite.FavoritesWsParameters.PARAM_COMPONENT;
+
+public class AddActionTest {
+  private static final String PROJECT_KEY = "project-key";
+  private static final String PROJECT_UUID = "project-uuid";
+  private static final int USER_ID = 123;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+  @Rule
+  public DbTester db = DbTester.create();
+  private DbClient dbClient = db.getDbClient();
+  private DbSession dbSession = db.getSession();
+
+  private FavoriteService favoriteService = new FavoriteService(dbClient, userSession);
+  private WsActionTester ws = new WsActionTester(new AddAction(userSession, dbClient, favoriteService, new ComponentFinder(dbClient)));
+
+  @Test
+  public void add_a_project() {
+    ComponentDto project = insertProjectAndPermissions();
+
+    TestResponse result = call(PROJECT_KEY);
+
+    assertThat(result.getStatus()).isEqualTo(HTTP_NO_CONTENT);
+    List<PropertyDto> favorites = dbClient.propertiesDao().selectByQuery(PropertyQuery.builder()
+      .setUserId(USER_ID)
+      .setKey("favourite")
+      .build(), dbSession);
+    assertThat(favorites).hasSize(1);
+    PropertyDto favorite = favorites.get(0);
+    assertThat(favorite.getResourceId()).isEqualTo(project.getId());
+    assertThat(favorite.getUserId()).isEqualTo(USER_ID);
+    assertThat(favorite.getKey()).isEqualTo("favourite");
```

Comment:
Nice to have : I think we could make such UT more readable using extracting().containsOnly() => 
```
assertThat(favorites)
   .extracting(PropertyDto::getResourceId, PropertyDto::getUserId, PropertyDto::getKey)
   .containsOnly(tuple(project.getId(), (long) USER_ID, "favourite"));
```